core/crypto/pbkdf2_hw.c

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/* Copyright (c) 2022, HiSilicon Limited. */
+
+#include <crypto/crypto.h>
+#include <crypto/pbkdf2_support.h>
+
+/* This is a hardware pbkdf2 function, without mac simulation */
+TEE_Result __weak hw_pbkdf2(uint32_t hash_id __unused,
```

Comment:
I don't think this weak function is needed, because when `CFG_CRYPTO_HW_PBKDF2=y` there will be a platform-specific implementation, and when `CFG_CRYPTO_HW_PBKDF2 != y` it is not used.